notifier/notifier.go

```diff
@@ -79,23 +74,26 @@ type Options struct {
 	RelabelConfigs []*config.RelabelConfig
 	// Used for sending HTTP requests to the Alertmanager.
 	Do func(ctx context.Context, client *http.Client, req *http.Request) (*http.Response, error)
-}
 
-// New constructs a new Notifier.
-func New(o *Options) *Notifier {
-	ctx, cancel := context.WithCancel(context.Background())
-
-	if o.Do == nil {
-		o.Do = ctxhttp.Do
-	}
+	Registerer prometheus.Registerer
+	Metrics    *AlertMetrics
+}
 
-	return &Notifier{
-		queue:  make(model.Alerts, 0, o.QueueCapacity),
-		ctx:    ctx,
-		cancel: cancel,
-		more:   make(chan struct{}, 1),
-		opts:   o,
+// AlertMetrics encapsulates all the metrics required for Notifications
+type AlertMetrics struct {
```

Comment:
Ok, so IIUC you are exporting this type now so that another user of this package could pass in their own metrics object. That's great. But they still wouldn't be able to actually set or Prom-register the fields of `AlertMetrics` because the fields are still unexported. So I guess they should be exported as well?

Or maybe even better instead of exporting those fields: export only `NewAlertMetrics()` - that way, a user will also not have to copy all the metrics definitions again and just call `NewAlertMetrics()` to set them all properly and then pass the result in.